docstore/doc.go

```diff
@@ -51,17 +51,66 @@
 // information.
 //
 //
-// Documents
+// Representing Documents
+//
+// A document is a set of named fields, each with a value. Values can themselves
+// be documents or lists of values.
 //
 // Docstore allows you to represent documents as either map[string]interface{} or
-// struct pointers. Using structs is recommended, because it enforces some structure
-// on your data. Docstore mimics the encoding/json package in its treatment of
+// struct pointers. When you represent a document as a map, the field are map keys
+// and the values are map values. For example, here is a document about a book
+// described as a map:
```

Comment:
In particular, would Author be a document because it is also a map?

Reply:
> What are the values in the example below?

I think the sentence just above, "When you represent a document as a map, the fields are map keys and the values are map values," says everything a Go programmer needs to understand which are fields and which are values.

> would Author be a document because it is also a map

Yes, I called this out in a new sentence.

Reply:
Good idea. Done. (I thought you might recognize the example. One of my top 3 favorite books.)